tests/Elrond/SignerTests.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include <gtest/gtest.h>
+#include "boost/format.hpp"
+
+#include "HexCoding.h"
+#include "PrivateKey.h"
+#include "PublicKey.h"
+#include "Elrond/Signer.h"
+#include "Elrond/Address.h"
+#include "TestAccounts.h"
+
+using namespace TW;
+using namespace TW::Elrond;
+
+
+TEST(ElrondSigner, Sign) {
+    auto input = Proto::SigningInput();
+    auto privateKey = PrivateKey(parse_hex(ALICE_SEED_HEX));
+    input.set_private_key(privateKey.bytes.data(), privateKey.bytes.size());
+
+    input.mutable_transaction()->set_nonce(0);
+    input.mutable_transaction()->set_value("0");
+    input.mutable_transaction()->set_sender(ALICE_BECH32);
+    input.mutable_transaction()->set_receiver(BOB_BECH32);
+    input.mutable_transaction()->set_gas_price(200000000000000);
+    input.mutable_transaction()->set_gas_limit(500000000);
+    input.mutable_transaction()->set_data("foo");
+
+    auto output = Signer::sign(input);
+    auto signature = output.signature();
+    auto signedTransaction = output.signed_transaction();
+    auto expectedSignature = "b88ad2fe98a7316ea432a0a76c18cd87200fe75f27a8f053ea6532b40317dbec5136c5463aef132ae951b7e60d45d921caaa5903e70821dcda98f237d4ec4308";
+    auto expectedSignedTransaction = (boost::format(R"({"nonce":0,"value":"0","receiver":"%1%","sender":"%2%","gasPrice":200000000000000,"gasLimit":500000000,"data":"foo","signature":"%3%"})") % BOB_BECH32 % ALICE_BECH32 % expectedSignature).str();
```

Comment:
is this a valid tx in block explorer?

Reply:
Yes, it is - the encoded transaction to be sent to the blockchain would look like this:

```
{
    "nonce": 0,
    "value": "0",
    "sender": "erd1l453hd0gt5gzdp7czpuall8ggt2dcv5zwmfdf3sd3lguxseux2fsmsgldz",
    "receiver": "erd1cux02zersde0l7hhklzhywcxk4u9n4py5tdxyx7vrvhnza2r4gmq4vw35r",
    "gasPrice": 200000000000000,
    "gasLimit": 500000000,
    "data": "foo",
    "signature": "b88ad2fe98a7316ea432a0a76c18cd87200fe75f27a8f053ea6532b40317dbec5136c5463aef132ae951b7e60d45d921caaa5903e70821dcda98f237d4ec4308"
}
```

The computed hash (returned upon invoking the API) is `cbecce3712064fe8b14bd668f02cf348d4ee5e4c73836d0a7e25482e5d0056d9`, and the link in block explorer would be:

https://explorer.elrond.com/transactions/cbecce3712064fe8b14bd668f02cf348d4ee5e4c73836d0a7e25482e5d0056d9

The link above is currently available (and will be until a hard reset of our testnet).

Does this answer the question?